generator/sources/java/src/com/kaltura/client/KalturaClientBase.java

```diff
@@ -100,16 +102,21 @@
 	private static final int RANDOM_SIZE = 16; 
 
 	private static final int MAX_DEBUG_RESPONSE_STRING_LENGTH = 1024;
-	protected KalturaConfiguration kalturaConfiguration;
-    protected List<KalturaServiceActionCall> callsQueue;
-    protected List<Class<?>> requestReturnType;
-    protected KalturaParams multiRequestParamsMap;
-	protected Map<String, Object> clientConfiguration = new HashMap<String, Object>();
-	protected Map<String, Object> requestConfiguration = new HashMap<String, Object>();
-
-	private static IKalturaLogger logger = KalturaLogger.getLogger(KalturaClientBase.class);
-    
-    private Header[] responseHeaders = null; 
+	
+	public static IKalturaLogger logger = KalturaLogger.getLogger(KalturaClientBase.class);
+	
+	private class ThreadInfo
+	{
+		public KalturaConfiguration kalturaConfiguration;
+		public List<KalturaServiceActionCall> callsQueue;
+		public List<Class<?>> requestReturnType = null;
+		public KalturaParams multiRequestParamsMap;
+		public Map<String, Object> clientConfiguration = new HashMap<String, Object>();
+		public Map<String, Object> requestConfiguration = new HashMap<String, Object>();
+		public Header[] responseHeaders;
+	}
+	
+    private ConcurrentHashMap<Thread, ThreadInfo> threadsInfo = new ConcurrentHashMap<Thread, ThreadInfo>(); 
```

Comment:
entries are only put into the map, however, are never removed from it (memory leak)